.github/workflows/main_debug.yml

```diff
@@ -1,27 +1,30 @@
 name: CI-Debug
 
-on:
+on: 
+# Runs on every push on master branch. If a push contains multiple commits, it will be ran on the latest one.
   push:
     # Runs on every push on master branch
     branches:
       - master
   pull_request:
     paths-ignore:
       - 'doc/**'
-
+    
 # Cancels running instances when a pull request is updated by a commit
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref || github.run_id }}
   cancel-in-progress: true
```

Comment:
Pretty sure it does, but have you try it?

Reply:
Yup, tried it, it works :)